client/src/components/Draftail/CommentableEditor/CommentableEditor.js

```diff
@@ -201,14 +237,45 @@ function CommentableEditor({commentApp, fieldNode, contentPath, rawContentState,
 
     return <DraftailEditor
     ref={editorRef}
-    onChange={setEditorState}
+    onChange={(editorState) => {
+      let newEditorState = editorState;
+      if (['undo', 'redo'].includes(editorState.getLastChangeType())) {
+        const filteredContent = filterInlineStyles(inlineStyles.map(style => style.type).concat(ids.map(id => COMMENT_STYLE_IDENTIFIER + id)), editorState.getCurrentContent());
+        newEditorState = forceResetEditorState(editorState, filteredContent);
+      }
+      setEditorState(newEditorState);
+    }
+    }
     editorState={editorState}
     controls={enabled ? [CommentControl] : []}
     decorators={enabled ? [{
-      strategy: findCommentStyleRanges,
+      strategy: (block, callback) => findCommentStyleRanges(block, callback),
       component: CommentDecorator 
     }] : []}
     inlineStyles={inlineStyles.concat(commentStyles)}
+    plugins={enabled ? [{
+      customStyleFn: (styleSet) => {
+        const commentStyles = styleSet.filter(style => style.startsWith(COMMENT_STYLE_IDENTIFIER));
+        const numStyles = commentStyles.count();
+        if (numStyles > 0) {
+          // There is at least one comment in the range
+          const commentIds = commentStyles.map(style => parseInt(style.slice(8)));
+          let background = '#01afb0';
+          if (commentIds.has(focusedId)) {
+            // Use the focused colour if one of the comments is focused
+            background = '#007d7e';
+          } else if (numStyles > 1) {
+            // Otherwise if we're in a region with overlapping comments, use a slightly darker colour than usual
+            // to indicate that
+            background = '#01999a';
```

Comment:
suggestion: use classes so that the colours are consistent and in one place. Especially now when the admin can have colour themes ([docs](https://docs.wagtail.io/en/stable/advanced_topics/customisation/admin_templates.html#custom-user-interface-colors))

Reply:
The issue with this is that as it's inside Draftail, we can't apply classes directly. (We can using a decorator, but due to the need to comment across entities, we can't guarantee a decorator will be used for the whole comment annotation)

That said, I do agree that these should be taken from somewhere, rather than hardcoded - as noted in the other PR, styling here is still placeholder, so this isn't something I've looked into in detail yet, how to retrieve these colours from the css,

Reply:
possibly just creating and retrieving the styles from an element with the right class...

Reply:
You should be able to replace these with CSS variables, we added them to Wagtail not so long ago.

Reply:
Okay great, this is something I'll look into then!